packages/interactivity/src/utils.ts

```diff
@@ -203,21 +228,31 @@ export function useCallback( callback, inputs ) {
  * available so functions like `getElement()` and `getContext()` can be used
  * inside the passed factory function.
  *
- * @param {Function} factory Imperative function that can return a cleanup
- *                           function.
- * @param {any[]}    inputs  If present, effect will only activate if the
- *                           values in the list change (using `===`).
+ * @param factory Imperative function that can return a cleanup
+ *                function.
+ * @param inputs  If present, effect will only activate if the
+ *                values in the list change (using `===`).
  */
-export function useMemo( factory, inputs ) {
+export function useMemo( factory: Function, inputs: any[] ) {
 	_useMemo( withScope( factory ), inputs );
```

Comment:
I think this function is completely wrong, fixing it here: https://github.com/WordPress/gutenberg/pull/60474

Let's revert these changes for now.